run_task.py

```diff
@@ -3,46 +3,84 @@
+                                 'allow_extra_args': True},
+               help='Run a task on vantage6 nodes. Optionally pass extra'
+                    'keyword arguments to be passed to the algorithm'
+                    '(i.e. "--key value")')
+@click.option('--method', default=DEFAULT_METHOD,
+              help='Method to run')
+@click.option('--image', default=DEFAULT_IMAGE,
+              help='Docker image to run')
+@click.option('--collaboration_id', default=DEFAULT_COLLABORATION_ID,
+              help='Identifier for collaboration to run task with', type=int)
+@click.option('--organization_ids', default=DEFAULT_ORGANIZATION_IDS,
+              help='Identifier for organization ids that should run task, '
+                   'pass as comma-separated string ("1,2")')
+@click.option('--username', default=DEFAULT_USERNAME)
+@click.option('--password', default=DEFAULT_PASSWORD)
+@click.option('--host', default=DEFAULT_HOST,
+              help='Host for vantage6 server')
+@click.option('--port', default=DEFAULT_PORT, type=int,
+              help='Port of vantage6 server')
+@click.option('--wait_time', default=DEFAULT_WAIT_TIME, type=int,
+              help='Time in seconds to wait in between polling tries')
+@click.option('--num_retries', default=DEFAULT_NUM_RETRIES, type=int,
+              help='Number of retries for polling task results')
+@click.option('--master/--rpc', default=True,
+              help='Whether to run master or RPC algorithm')
+@click.pass_context
+def main(context, method: str, image: str, collaboration_id: int,
+         organization_ids: str, master: bool, username: str, password: str,
+         host: str, port: int, wait_time: int, num_retries: int):
+    # context.args collects unkown arguments in a list:
+    # (['--unknown_var', 'value3', '--unknown_var2', 'value4'])
+    kwargs = {context.args[i][2:]: context.args[i + 1]
+              for i in range(0, len(context.args), 2)}
+
+    organization_ids = [int(organization_id) for organization_id in organization_ids.split(',')]
+
+    client = vtgclient.Client(host, port)
+    client.authenticate(username, password)
     client.setup_encryption(None)
 
-    task = client.post_task(name=METHOD, image=IMAGE, collaboration_id=COLLABORATION_ID,
-                            organization_ids=ORGANIZATION_IDS,
-                            input_={'method': METHOD, 'master': MASTER, 'kwargs': {'exclude_orgs': ORGANIZATION_IDS}})
+    task = client.post_task(name=method,
+                            image=image,
+                            collaboration_id=collaboration_id,
+                            organization_ids=organization_ids,
+                            input_={'method': method,
+                                    'master': master,
+                                    'kwargs': kwargs})
 
     print(task)
     results = []
 
-    for i in range(RETRIES):
+    for i in range(num_retries - 1):
```

Comment:
Hm why `num_retries - 1`? If you specify 1 for `num_retries` this loop will never run (because `range(0)` is empty).
Is this maybe about the semantics of the word retries? Like, when you try once you have zero retries?

If this is the case I'd just rename `num_retries` to `num_tries`, or you could do `num_retries + 1` to be semantically correct about retries.

Reply:
Hmmm, yeah it is about the semantics, but indeed this fails. I will rename it to `num_tries` because I think that is what is happening here.